### PR TITLE
Set up golangci-lint

### PR DESCRIPTION
**Pull request by shuheiktgw**
I've set up golangci-lint and `make fmt` task

### CODE REVIEW

File: compiler/compiler.go [hunk 1]
```diff
@@ -87,7 +88,10 @@ const (
 
 // Compile compile the target Protocol Buffers file and produces all file descriptors.
 func (c *Compiler) Compile(ctx context.Context, file *source.File, opts ...Option) ([]*descriptorpb.FileDescriptorProto, error) {
-	c.importPaths = c.importPaths[:]
```
Comment (shuheiktgw) [inline compiler/compiler.go:90]: I didn't know that either, but unlike Python, `c.importPaths[:]` does not create a copy of a slice...

File: generator/config.go [hunk 2]
```diff
@@ -16,9 +16,9 @@ type Config struct {
 	// Option specify options to be passed gRPC Federation code generator.
 	Option *PluginOption `yaml:"opt"`
 	// AutoProtocGenGo automatically run protoc-gen-go at the time of editing proto. default is true.
-	AutoProtocGenGo *bool `yaml:"auto-protoc-gen-go"`
+	AutoProtocGenGo *bool `yaml:"autoProtocGenGo"`
 	// AutoProtocGenGoGRPC automatically run protoc-gen-go-grpc at the time of editing proto. default is true.
-	AutoProtocGenGoGRPC *bool `yaml:"auto-protoc-gen-go-grpc"`
```
Comment (shuheiktgw) [inline generator/config.go:21]: The linter raises an error to a snake-cased tag, but if it is intentional, I'll return it.

File: resolver/resolver_test.go [hunk 3]
```diff
@@ -2103,9 +2103,6 @@ func TestLiteral(t *testing.T) {
 					Build(t),
 			).SetAlias(ref.Field(t, "content", "Content", "message_field")).Build(t),
 		},
-	)
-	content.Fields = append(
```
Comment (shuheiktgw) [inline resolver/resolver_test.go:2107]: The diff looks a little confusing, but it just merges the two `append` into one.

File: generator/config.go [hunk 4]
```diff
@@ -77,7 +77,7 @@ func setupConfig(cfg *Config) error {
 	pluginNameMap := make(map[string]struct{})
 	for _, plugin := range cfg.Plugins {
 		name := plugin.Plugin
-		if !strings.HasPrefix("protoc-gen-", name) {
+		if !strings.HasPrefix(name, "protoc-gen-") {
```
Comment (shuheiktgw) [inline generator/config.go:80]: 👀

**Review by goccy — APPROVED** [review]
LGTM 👍

**Pull request merged**